packages/x-charts-pro/src/context/ZoomProvider/useSetupPan.ts

```diff
@@ -42,25 +40,53 @@ export const useSetupPan = () => {
 
       const point = getSVGPoint(element, event);
       const movementX = point.x - touchStartRef.current.x;
-
-      const max = touchStartRef.current.maxX;
-      const min = touchStartRef.current.minX;
-      const span = max - min;
-
-      let newMinRange = min - (movementX / area.width) * span;
-      let newMaxRange = max - (movementX / area.width) * span;
-
-      if (newMinRange < MIN_RANGE) {
-        newMinRange = MIN_RANGE;
-        newMaxRange = span;
-      }
-
-      if (newMaxRange > MAX_RANGE) {
-        newMaxRange = MAX_RANGE;
-        newMinRange = MAX_RANGE - span;
-      }
-
-      setZoomRange([newMinRange, newMaxRange]);
+      const movementY = (point.y - touchStartRef.current.y) * -1;
+
+      const newZoomData = touchStartRef.current.zoomData.map((zoom) => {
+        const option = options[zoom.axisId];
+        if (!option || !option.panning) {
+          return zoom;
+        }
+
+        const min = zoom.start;
+        const max = zoom.end;
+        const span = max - min;
+        const MIN_PERCENT = option.start;
+        const MAX_PERCENT = option.end;
```

Comment:
What about `minStart` and `maxEnd` ? I was confused about what those values were for

Reply:
Do you mean to change the option to `option.minStart`?